src/rust/src/declarative_asn1/types.rs

```diff
@@ -70,6 +73,34 @@ impl Annotation {
     }
 }
 
+#[derive(pyo3::FromPyObject)]
+#[pyo3::pyclass(frozen, module = "cryptography.hazmat.bindings._rust.asn1")]
+pub struct PrintableString {
+    pub(crate) inner: pyo3::Py<pyo3::types::PyString>,
+}
+
+#[pyo3::pymethods]
+impl PrintableString {
+    #[new]
+    #[pyo3(signature = (inner,))]
+    fn new(inner: pyo3::Py<pyo3::types::PyString>) -> Self {
```

Comment:
We should validate that `inner` is a valid PrintableString, rather than defering that to serialization

Reply:
Should I replicate the logic from https://github.com/alex/rust-asn1/blob/7ff2acc222bf333d75ddda14c6bb0dc6704f5d8d/src/types.rs#L402 ? We could also make `verify` public.

Reply:
`asn1::PrintableString::new(s).is_none()` should be enough to check if its invalid

Reply:
Or call rust-asn1's `PrintableString::new` and discard the result